### PR TITLE
Email-less way to contact tutees/tutors

### PR DESCRIPTION
**Pull request by realmayus**
Closes https://github.com/corona-school/project-user/issues/288.

As per #288, this PR aims to remove the need of exposing a course's participants' emails to the instructors for privacy reasons. It removes the email addresses from the list and provides a way of selecting individual students for sending a message.
It amends the previous participant selection feature by adding a second button for selecting individual participants to contact.
Additionally, the contact modal has been revamped.

This goes hands in hands with this [PR](https://github.com/corona-school/backend/pull/230) in the backend repo which drops the email addresses from the payload and allows clients to specify a subset of users to contact when reaching out to the groupMail endpoint.

### CODE REVIEW

File: src/components/Modals/ContactCourseModal.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  setSelectedParticipants: (selectedParticipants: CourseParticipant[]) => void;
+  setSelectingParticipants: (isSelecting: boolean) => void;
+  type: 'instructorToParticipants' | 'participantToInstructors';
+}
+
+const ContactCourseModal: React.FC<Props> = ({
+  course,
+  setSelectedParticipants,
+  selectedParticipants,
+  type,
```
Comment (Jonasdoubleyou) [inline src/components/Modals/ContactCourseModal.tsx:21]: I'm not quite sure if this should really be one component, especially as for the "participantToInstructors" the selectedParticipants is useless. Maybe it makes sense to split this up in three components, `<ContactForm title=... description=... submit={() => ...} />` and then two wrappers for that for the two usecases.

**Review by Jonasdoubleyou — APPROVED** [review]
Looks good, haven't tested it as the review app is down and we need a different backend anyways, so this should be tested in dev before prod deployment ...

**Review by Jonasdoubleyou — APPROVED** [review]

**Comment by realmayus** [comment]
Yeah, I'm currently working on the backend

File: src/routes/CourseDetail.tsx [hunk 2]
```diff
@@ -358,22 +371,34 @@ const CourseDetail = (props: Props) => {
...
+  const hasEnded = () => {
+    if (getLastLectureEnd) {
+      return moment().isAfter(getLastLectureEnd());
     }
 
     return false;
   };
 
+  const canContact = () => {
+    if (getLastLectureEnd) {
```
Comment (Jonasdoubleyou) [inline src/routes/CourseDetail.tsx:396]: ts ts ts ... who reviewed this ;) ... https://github.com/corona-school/web-user-app/commit/af01ff6470341db6cf0175e1a8536bb2b9ccf98a

**Pull request merged**